### PR TITLE
Ft/event-card

### LINKED ISSUE #321: @/charterafrica/components/EventCard
### Description

Reusing code (or knowledge) from Spotlight and Post card components, implement the EventCard component to be used in events [section](https://www.figma.com/file/gbnaO1FYPq16V2gAFbLOkl/CfA---EPD-Charter-Africa---Sep-2022?node-id=9108%3A28134&t=gHql57tAK6HPaLGe-0).

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description
This PR implements `Event` Card as per the design

Fixes #321

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
screencapture-localhost-3000-opportunities-fellowships-2023-03-01-09_56_28

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/charterafrica/src/components/EventsFellowshipsAndGrants/EventsFellowshipsAndGrants.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { Section } from "@commons-ui/core";
+import { Typography, Box, Divider } from "@mui/material";
+import React from "react";
+
+import OpportunityCardList from "../OpportunityCardList";
```
Comment (kilemensi) [inline apps/charterafrica/src/components/EventsFellowshipsAndGrants/EventsFellowshipsAndGrants.js:5]: No relative import in Next.js side of things.

File: apps/charterafrica/src/components/EventsFellowshipsGrantsHeader/EventsFellowshipsGrantsHeader.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ import React from "react";
 
 import { secondary } from "@/charterafrica/colors";
 
-const GrantsFellowshipsHeader = React.forwardRef(
-  function GrantsFellowshipsHeader(props, ref) {
+const EventsFellowshipsGrantsHeader = React.forwardRef(
```
Comment (kilemensi) [inline apps/charterafrica/src/components/EventsFellowshipsGrantsHeader/EventsFellowshipsGrantsHeader.js:7]:
```suggestion
const FundingHeader = React.forwardRef(
```

Or `OpportunityHeader` should be enough

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js [hunk 3]
```diff
@@ -67,20 +71,50 @@ const OpportunityCard = React.forwardRef(function OpportunityCard(props, ref) {
...
+              <Link
+                color="neutral.dark"
+                href={registerLink?.href}
+                underline="always"
+                variant="caption"
+                sx={{
+                  textTransform: "capitalize",
+                }}
+              >
+                Register here
```
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:112]: ?

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListGrid.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Grid } from "@mui/material";
+import React from "react";
+
+import OpportunityCard from "@/charterafrica/components/OpportunityCard";
+
+const OpportunityCardListGrid = React.forwardRef(
+  function OpportunityCardListGrid(props, ref) {
+    const { items, config, showAll } = props;
+
+    return (
```
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListGrid.js:10]: If it's a standalone component, always check if we have `items`

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListGrid.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { Grid } from "@mui/material";
+import React from "react";
+
+import OpportunityCard from "@/charterafrica/components/OpportunityCard";
+
+const OpportunityCardListGrid = React.forwardRef(
```
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListGrid.js:6]: ListGrid? Why not just `OpportunityCardGrid`?

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js [hunk 6]
```diff
@@ -1,7 +1,10 @@
...
+import { Typography, Box, Button } from "@mui/material";
 import React, { useState } from "react";
 
-import OpportunityCard from "../OpportunityCard";
+import FeaturedPostCard from "../FeaturedPostCard/FeaturedPostCard";
+
+import OpportunityCardListGrid from "./OpportunityCardListGrid";
+import OpportunityCardListHeader from "./OpportunityCardListHeader";
 
 const OpportunityCardList = React.forwardRef(function OpportunityCardList(
```
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:9]: Why not just Opportunities or OpportunityCards, etc. Is it really a list.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListHeader.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            }}
+            onClick={onClick}
+          >
+            <Typography
+              color="neutral.dark"
+              variant="p3SemiBold"
+              sx={{
+                textDecoration: "underline",
+              }}
+            >
```
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardListHeader.js:51]: You can set `color` and `typography` on Button using `sx`.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 8]
```diff
@@ -121,13 +129,15 @@ export async function processPageFellowships({ blocks }) {
         updatedAt: "2023-01-26T11:21:14.868Z",
         url: "/images/charter-africa-brand.svg",
       },
-      deadline: "2023-02-11",
+      link: {
+        href: `/grants/${i}`,
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:133]: ?

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---


Needs a bit of clean up before a review.

File: apps/charterafrica/payload.config.ts [hunk 9]
```diff
@@ -36,13 +36,15 @@ dotenv.config({ path: path.resolve(__dirname, "./.env.local") });
 
 const appURL = process.env.PAYLOAD_PUBLIC_APP_URL;
 
-const cors = process?.env?.PAYLOAD_CORS?.split(",")
-  ?.map((d) => d.trim())
-  ?.filter(Boolean) ?? [];
+const cors =
+  process?.env?.PAYLOAD_CORS?.split(",")
+    ?.map((d) => d.trim())
```
Comment (koechkevin) [inline apps/charterafrica/payload.config.ts:41]: Can process be null? I doubt.

File: apps/charterafrica/src/components/EventsFellowshipsAndGrants/EventsFellowshipsAndGrants.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    const itemsByStatus = items.reduce((acc, item) => {
+      const { status } = item;
+      acc[status] = acc[status] || [];
+      acc[status].push(item);
+      return acc;
+    }, {});
+
+    const itemsByStatusArray = Object.keys(itemsByStatus).map((key) => {
+      return {
+        title: `${key}`,
```
Comment (koechkevin) [inline apps/charterafrica/src/components/EventsFellowshipsAndGrants/EventsFellowshipsAndGrants.js:26]:
```suggestion
        title:  key,
```

File: apps/charterafrica/payload.config.ts (hunk 9 above)
Comment (kelvinkipruto) [reply apps/charterafrica/payload.config.ts]: Yes. This caused the server not to start f null

File: apps/charterafrica/src/lib/data/common/index.js (hunk 8 above)
Comment (kelvinkipruto) [reply apps/charterafrica/src/lib/data/common/index.js:133]: @kilemensi Not sure what the issue is here

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js [hunk 11]
```diff
@@ -67,20 +80,50 @@ const OpportunityCard = React.forwardRef(function OpportunityCard(props, ref) {
...
+          <RichText
             color="neutral.dark"
-            variant="caption"
-            sx={{
-              textTransform: "uppercase",
-            }}
-          >
-            {config.deadlineText}: {deadline}
-          </Typography>
+            lineClamp="3"
```
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:85]:
```suggestion
            lineClamp={3}
```

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js [hunk 12]
```diff
@@ -67,20 +80,50 @@ const OpportunityCard = React.forwardRef(function OpportunityCard(props, ref) {
...
+            variant="p1"
+            elements={excerpt}
```
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:87]: Also add `height` or `maxHeight` style

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js [hunk 13]
```diff
@@ -67,20 +80,50 @@ const OpportunityCard = React.forwardRef(function OpportunityCard(props, ref) {
...
+              color="neutral.dark"
+              variant="caption"
+              sx={{
+                textTransform: "uppercase",
+              }}
+            >
+              {config?.dateText ? `${config?.dateText}:` : null} {date}
+            </Typography>
+          </Grid>
+          {registerLink ? (
```
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:110]: Check if  `registerLink?.href` exists.

File: apps/charterafrica/src/components/OpportunityCards/OpportunityCards.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+      }}
+    >
+      <OpportunityCardHeader
+        title={title}
+        onClick={() => setShowAll(!showAll)}
+        showAll={showAll}
+        showAllText={config?.showAllText || "Show all"}
+        showLessText={config?.showLessText || "Show less"}
+      />
+      {featuredEvent ? <FeaturedPostCard {...featuredEvent} /> : null}
```
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCards/OpportunityCards.js:35]: This is equivalent to conditional rendering  `{featuredEvent  && <FeaturedPostCard {...featuredEvent} />}`

**Comment by kilemensi** [comment]
Can you address the review above @kelvinkipruto ? We need to deploy this today.

PS: I also see you've marked some of my previous comments are resolved but I don't see any changes e.g. https://github.com/CodeForAfrica/ui/pull/343#discussion_r1121500115

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js (hunk 12 above)
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:87]: @koechkevin Why add heigth? isn't it handled by the variant?

**Comment by kelvinkipruto** [comment]
> Can you address the review above @kelvinkipruto ? We need to deploy this today.
> 
> PS: I also see you've marked some of my previous comments are resolved but I don't see any changes e.g. [#343 (comment)](https://github.com/CodeForAfrica/ui/pull/343#discussion_r1121500115)

@kilemensi I moved it [OpportunityHeader](https://github.com/CodeForAfrica/ui/blob/d6d7e1b305d851aae5572716f3a109f73fabe4fe/apps/charterafrica/src/components/OpportunityHeader/OpportunityHeader.js#L7-L9) as suggested

**Comment by kilemensi** [comment]
> > Can you address the review above @kelvinkipruto ? We need to deploy this today.
> > PS: I also see you've marked some of my previous comments are resolved but I don't see any changes e.g. [#343 (comment)](https://github.com/CodeForAfrica/ui/pull/343#discussion_r1121500115)
> 
> @kilemensi I moved it [OpportunityHeader](https://github.com/CodeForAfrica/ui/blob/d6d7e1b305d851aae5572716f3a109f73fabe4fe/apps/charterafrica/src/components/OpportunityHeader/OpportunityHeader.js#L7-L9) as suggested

Cool @kelvinkipruto ... I still see a lot of EventsGrantsAndFellowship naming on review page.

Screenshot 2023-03-02 at 13 27 34

**Comment by kelvinkipruto** [comment]
> Cool @kelvinkipruto ... I still see a lot of EventsGrantsAndFellowship naming on review page.
> 
@kilemensi Updated it to `OpportunityPage`

**Review by koechkevin — APPROVED** [review]

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**